src/coreclr/gc/gc.cpp

```diff
@@ -52176,6 +52186,19 @@ void CFinalize::WalkFReachableObjects (fq_walk_fn fn)
     }
 }
 
+void CFinalize::LogCounts(const char* label)
+{
+    size_t genCounts[max_generation + 1] = {};
+    for (int gen = 0; gen <= max_generation; ++gen)
+    {
+        unsigned int seg = gen_segment(gen);
+        genCounts[gen] = SegQueueCount(seg);
+    }
+    dprintf(1, ("Finalizer counts @%5s; 0: %6zu; 1: %6zu, 2: %6zu; CF: %6zu; F: %6zu; Free: %6zu",
+        label, genCounts[0], genCounts[1], genCounts[2],
+        SegQueueCount(CriticalFinalizerListSeg), SegQueueCount(FinalizerListSeg), SegQueueCount(FreeListSeg)));
+}
```

Comment:
Existing event only records how much is added to the finalization list.  This lets us watch the size over time.  I think this is probably worth a new event as well, but I haven't decided what it should be.  (perhaps as minimal as just the "begin" ones but only if something is non-zero?)

Reply:
probably shouldn't be dprintf(1)

Reply:
#ifdef

Reply:
add event for beginning of GC - add to HeapStats?